share_extension/components/content_view/content_view.tsx

```diff
@@ -0,0 +1,89 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React, {useEffect} from 'react';
+import {View} from 'react-native';
+import {KeyboardAwareScrollView} from 'react-native-keyboard-aware-scroll-view';
+
+import {setShareExtensionUserAndChannelIds, useShareExtensionState} from '@share/state';
+import {changeOpacity, makeStyleSheetFromTheme} from '@utils/theme';
+
+import Attachments from './attachments';
+import LinkPreview from './link_preview';
+import Message from './message';
+import Options from './options';
+
+import type {Database} from '@nozbe/watermelondb';
+
+type Props = {
+    currentChannelId: string;
+    currentUserId: string;
+    database: Database;
+    theme: Theme;
+}
+
+const getStyles = makeStyleSheetFromTheme((theme: Theme) => ({
+    container: {
+        flex: 1,
+    },
+    content: {
+        paddingTop: 20,
+    },
+    divider: {
+        backgroundColor: changeOpacity(theme.centerChannelColor, 0.08),
+        height: 1,
+        marginBottom: 8,
+        marginHorizontal: 20,
+        marginTop: 20,
+    },
+}));
+
+const ContentView = ({database, currentChannelId, currentUserId, theme}: Props) => {
+    const {linkPreviewUrl, files, serverUrl, channelId} = useShareExtensionState();
+    const styles = getStyles(theme);
+
+    useEffect(() => {
+        setShareExtensionUserAndChannelIds(currentUserId, currentChannelId);
```

Comment:
This seems out of the ordinary. The current channel id that comes from the props is the one from the main app, right? So changes in the main app are making a change here, right? Not sure when do we expect this currentChannelId to change (websocket event?) and why it should change the share extension state.

Reply:
this is totally expected as the first time you open the extension it should allow you to directly post to the channel you are in or last visited in the main app, same as with the server.

also this only gets executed on mount and not after even if it changes on the main app

Reply:
How is that? If I read the code correctly (I may have missed something) the `currentChannelId` is brought from the database (an observable), and the dependencies of this effect include the `currentChannelId` and the `currentUserId`, so any change on those will re-trigger the effect, right?

Reply:
in fact you are right, if I use the main app and switch a channel while the extension is opened, the selected channel will change. Not sure if that is a problem though

Reply:
Not so worried about the "actively going to the app and switch channels" (which seems doable but I don't think will happen often) but about a websocket event moving you to a different channel. It is also a really limited timeframe (before the websocket closes once the app is sent to the background) and really specific case (you are kicked out of your current channel).

Probably not a big deal right now, but I guess it is worth creating a ticket (and maybe open it for help wanted once we have a clear idea of the behaviour we want?)

Reply:
I mean, the only way I can think of that happening is if you are removed from a channel just while you were going to share some content and the WS is still connected. I'd say we should build that bridge when the road is actually broken